### PR TITLE
Nonstatic Backward Pass

### PR DESCRIPTION
**Pull request by tpunnoose**
This fixes improper variable names for the case when static_bp = false. 

It also makes the default iteration count in solver.stats 1 instead of 0 to prevent indexing error (this makes the tests fail because the expect one iteration less).

### CODE REVIEW

File: src/solver_opts.jl [hunk 1]
```diff
@@ -156,7 +156,7 @@ end
 
 function reset!(stats::SolverStats, N::Int=0)
     stats.is_reset && return nothing
-    stats.iterations = 0
+    stats.iterations = 1
```
Comment (bjack205) [inline src/solver_opts.jl:159]: Why did you change this? This should be at zero so that the iteration counts work correctly.

**Comment by bjack205** [comment]
In general, make sure your PR passes the tests before submitting. You can do this using
```julia
] test Altro
```
in your REPL.

File: src/solver_opts.jl (hunk 1 above)
Comment (tpunnoose) [reply src/solver_opts.jl:159]: Yeah I wasn't 100% sure about the best way to fix this, but stats.iteration is used as an array index in multiple places so you get indexing errors when it starts at 0.

Comment (bjack205) [reply src/solver_opts.jl:159]: Have you run into this in practice? It shouldn't ever try to index into the `stats` if it hasn't completed an iteration. I'm more inclined to find where it's trying to index when it shouldn't be, since that's a problem. Changing this won't fix the underlying issue, just prevent an error (basically making the bug harder to detect).

Comment (tpunnoose) [reply src/solver_opts.jl:159]: yes see https://github.com/RoboticExplorationLab/Altro.jl/issues/4

**Comment by bjack205** [comment]
Closing this since this being fixed in the new version #31
